### DISCUSSION TITLE
Disconnections only when using the container on Docker host, not as a HA addon.

### DISCUSSION THREAD

**Discussion by Mincka** [mrlt8/docker-wyze-bridge#761] [General]
Hi,

I am using this container as HA addon for a while and it works quite great. I have HA OS on HA Blue and a few addons.
I have disconnections from time to time but that's very rare.

I'm in the process of moving out HA addons that are not tightly coupled to HA on a different setup in my network.
It's a NUC that's running few LXC containers and VMs. It's way more powerful (Celeron N5105 with 8 GB or ram) that the ODroidN2+ of HA Blue.

One of the LXC is a Docker host where I run Frigate for a few Hikvision cameras just fine.

So I tried to move `wyze-bridge` from HA to this Docker host  with the same configuration.
Frigate and `wyze-bridge` then are "sibling" containers on the same host and Frigate is the main client of `wyze-bridge`.

But the connections are very unstable, I see lots of disconnections. Almost immediately after the first connection.

I thought this could be related to a bandwidth issue since I also host Frigate on the same docker host / NUC, but that's not the case since I stopped this addon and the issues are the same.

I checked for other bottlenecks at network, CPU or memory levels, but I did not see anything interesting.
The only thing that changes from one setup to another is that the cameras are not in the same subnet than before.
My docker host is on 192.168.0.0/24, but HA and the cameras are on 10.0.0.0/24.
Not sure if that's relevant since my Frigate container on the Docker host works just fine the Hikvision NVR that's also 10.0.0.0/24.

HA Host Config
```
WYZE_EMAIL: xxx
WYZE_PASSWORD: xxx
NET_MODE: ANY
SNAPSHOT: API
MQTT_DTOPIC: homeassistant
ENABLE_AUDIO: false
SUBSTREAM: false
CAM_OPTIONS: []
ON_DEMAND: false
QUALITY: HD60
FILTER_BLOCK: true
RTSP_PATHS_ALL_READUSER: julien
RTSP_PATHS_ALL_READPASS: julien
FRESH_DATA: true
```

Docker Compose on Docker Host
```
version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest
        ports:
            - 1935:1935 # RTMP
            - 8555:8554 # RTSP
            - 8888:8888 # HLS
            - 8889:8889 #WebRTC
            - 8189:8189/udp # WebRTC/ICE
            - 5001:5000 # WEB-UI
        environment:
            - WYZE_EMAIL=xxx
            - WYZE_PASSWORD=xxx
            - NET_MODE=ANY
            - WB_IP=192.168.0.4
            - MQTT_HOST=False
            - FRESH_DATA=1
            - TZ=Europe/Paris
            - QUALITY=HD60
```
Log up/down for a single camera (others were filtered out):
```
15:02:51 [INFO][WyzeBridge] ✅ '/garage stream is UP! (3/3)
15:02:51 [INFO][WyzeBridge] 📖 New client reading from garage
2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49720] [c->s] TEARDOWN rtsp://0.0.0.0:8554/garage/ RTSP/1.0
CSeq: 5
Session: 2831126a-f1c7-4f27-af7b-7831f3bdd84a
User-Agent: Lavf59.16.100


2023/04/02 15:02:53 INF [RTSP] [session dfd1844d] runOnRead command stopped
2023/04/02 15:02:53 INF [RTSP] [session dfd1844d] destroyed (torn down by 127.0.0.1:49720)
2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49720] [s->c] RTSP/1.0 200 OK
CSeq: 5
Server: gortsplib


2023/04/02 15:02:53 INF [RTSP] [conn 127.0.0.1:49720] closed (EOF)
15:02:53 [INFO][WyzeBridge] 10.0.0.75 - - [02/Apr/2023 15:02:53] "GET /snapshot/garage.jpg?1680440568991 HTTP/1.1" 200 -
2023/04/02 15:02:53 INF [RTSP] [conn 127.0.0.1:49750] opened
2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [c->s] OPTIONS rtsp://0.0.0.0:8554/garage RTSP/1.0
CSeq: 1
User-Agent: Lavf59.16.100


2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [s->c] RTSP/1.0 200 OK
CSeq: 1
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib


2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [c->s] DESCRIBE rtsp://0.0.0.0:8554/garage RTSP/1.0
Accept: application/sdp
CSeq: 2
User-Agent: Lavf59.16.100


2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [s->c] RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://0.0.0.0:8554/garage/
Content-Length: 291
Content-Type: application/sdp
Server: gortsplib

v=0
o=- 0 0 IN IP4 127.0.0.1
s=Stream
c=IN IP4 0.0.0.0
t=0 0
m=video 0 RTP/AVP 96
a=control:mediaUUID=c2016ab4-9117-4c4e-86e9-db381555a928
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1; sprop-parameter-sets=Z01AKJ2oHgCJ+WEAAAMAAQAAAwAehA==,aO48gA==; profile-level-id=4D4028

2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [c->s] SETUP rtsp://0.0.0.0:8554/garage/mediaUUID=c2016ab4-9117-4c4e-86e9-db381555a928 RTSP/1.0
CSeq: 3
Transport: RTP/AVP/TCP;unicast;interleaved=0-1
User-Agent: Lavf59.16.100


2023/04/02 15:02:53 INF [RTSP] [session d8418c0a] created by 127.0.0.1:49750
2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [s->c] RTSP/1.0 200 OK
CSeq: 3
Server: gortsplib
Session: e24977d1-4d15-4839-b8c4-afd1c464cfd5
Transport: RTP/AVP/TCP;unicast;interleaved=0-1;ssrc=E9A7BBAD


2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [c->s] PLAY rtsp://0.0.0.0:8554/garage/ RTSP/1.0
CSeq: 4
Range: npt=0.000-
Session: e24977d1-4d15-4839-b8c4-afd1c464cfd5
User-Agent: Lavf59.16.100


2023/04/02 15:02:53 INF [RTSP] [session d8418c0a] is reading from path 'garage', with TCP, 1 track (H264)
2023/04/02 15:02:53 INF [RTSP] [session d8418c0a] runOnRead command started
2023/04/02 15:02:53 DEB [RTSP] [conn 127.0.0.1:49750] [s->c] RTSP/1.0 200 OK
CSeq: 4
RTP-Info: url=rtsp://0.0.0.0:8554/garage/mediaUUID=c2016ab4-9117-4c4e-86e9-db381555a928;seq=1485;rtptime=2586622032
Server: gortsplib
Session: e24977d1-4d15-4839-b8c4-afd1c464cfd5


15:02:53 [INFO][WyzeBridge] 📖 New client reading from garage
15:02:53 [INFO][WyzeBridge] 📕 Client stopped reading from garage
```

**Comment by mrlt8** [comment]
Starting with v2, we've switched to on-demand connections to the camera instead of a persistent connection so that the bridge will disconnect from the camera if there aren't any clients connected.

Does it still disconnect if you try to view the WebRTC stream?

**Reply by Mincka** [reply]
Indeed, that's what I guess I'm seeing. I get the "Disconnected" pop-ups but it seems to be only a behavior in the GUI.
The feeds look stable for Frigate so it seems that's just a "feeling" because of the pop-ups in the GUI and the messages in the logs, that are related to the web GUI as a client, not Frigate.
I need to test a little more to compare both versions (HA Addon and Docker) but I don't think there's an issue in the end.
I'll mark this answer as a solution if I can confirm this.

Thank you for you reply and thanks again for this great project!

**Reply by Mincka** [reply]
I can confirm than when using the container on my Docker host with WebRTC streams running in `wyze-bridge` GUI, I don't get any of the "Disconnected" pop-ups.

In Snapshot mode, I get the Disconnected pop ups about 5 to 10 seconds after the Connected pop-ups.
![image](https://user-images.githubusercontent.com/6184289/229377680-9a06c646-daae-4d1e-a416-f0f7c88ee88f.png)

I only get them on my Docker host with the following error messages:
![image](https://user-images.githubusercontent.com/6184289/229377747-b49cc412-a0b5-4720-a0b3-fc014a36c8f1.png)

Nothing like this in the HA Addon.
It looks like the Docker version has a local ffmpeg process (`User-Agent: Lavf59.16.100`?) that is constantly trying to connect and stops.

**Reply by mrlt8** [reply]
This is probably related to the fix for #750 where the bridge will now connect to all enabled cameras to update the snapshot/preview in the WebUI.

Do you still get that message if you close the WebUI or set the refresh time to 0? You could also disable any cameras you're not using to prevent the snapshots from updating.

**Reply by Mincka** [reply]
I am unable to reproduce today. Yesterday, I had issues to set my NVR to the new wyze_bridge feeds.
I don't why but the NVR was unable to authenticate on the container and I did nothing else than just replicating "RTSP_" settings.
I even confirmed it was working fine on VLC but no more for the container feeds vs HA addon.
Anyway, I removed the `- RTSP_PATHS_ALL_READPASS=xxx` and removed the authentication for the wyze_bridge feeds on my NVR, that is now able to be the one client always connected to the wyze_bridge.
That why I think I don't get any errors today. I don't want to bother more since it's working fine now.
Thank you again!
